### ISSUE TITLE
need a nice installation

### ISSUE THREAD

**Issue by JohnHerry** [TimDettmers/bitsandbytes#1248]
### Feature request

I have a env of python3.10, torch2.3.0, cuda12.1,  and when I install this package :  
pip install bitesandbytes  
it will force reinstall my pytorch to torch2.3.1.  I think this package should gave a compatible design with the local pytorch version. at least, there should better be a document that can tell the Correspondence relationship about torch version and this bitsandbytes versions.

 

### Motivation

to get a smooth package installation

### Your contribution

no

**Comment by younesbelkada** [comment]
How do you install bnb? AFAIK when you do `pip install bitsandbytes` it won't re-install `torch` unless you do `pip install -U bitsandbytes`

**Comment by JohnHerry** [comment]
Yes, I installed with --force-reinstall, thank you.

**Comment by JohnHerry** [comment]
by the way, how to install the bitsandbytes with local cuda version? My test of :  
python -m bitsandbytes  
got the info:  
CUDA SETUP: Required library version not found: libbitsandbytes_cuda121.so.   
I would better not recompile from source. it is in the docker image build on Dockerfile.

**Comment by JohnHerry** [comment]
> Thanks @JohnHerry ! You can follow the instructions here: https://huggingface.co/docs/bitsandbytes/main/en/installation?source=Linux there should be a flag to specify the cuda version I belive, @matthewdouglas what is the way to provide a specific cuda version when compiling from source using CMake?

Yes, It may need to compile the libbitsandbytes_cuda121.so from c source. I tried to install with python source:  
python setup.py install   
got the same library package not found error.

**Comment by JohnHerry** [comment]
> by the way, how to install the bitsandbytes with local cuda version? My test of : python -m bitsandbytes got the info: CUDA SETUP: Required library version not found: libbitsandbytes_cuda121.so. I would better not recompile from source. it is in the docker image build on Dockerfile.

Sloved!
I see in the docker container, that there is only a libbitsandbytes_cuda118.so installed. while in my another conda env, my bitsandbytes installation there are a batch of  .so versions corresponding different cuda versions.  I copied the libbitsandbytes_cuda121.so into the docker container by hand, and run `python -m bitsandbytes`,  it report another error:  
```
`CXXABI_1.3.9' not found   
```  
then I noticed that the docker container is based on CentOS7 image and the libstdc++.so is too old. I think that the bitesandbytes installer may detected this and installed a very old version of  so.    After I repalce the libstdc++.so to a newer version. the package install test is ok now.
